### PR TITLE
Correction of an issue #337

### PR DESCRIPTION
**Pull request by RestorerZ**
Added copying libsodium.dll to the top-level directory in the appveyor.bat file. Which fixes the situation pointed out in this issue #337 .

In addition, the following changes have been made:

Using, where possible, tools and features available in Windows instead of Cygwin tools.

Added an exception file when creating an archive file. As much as possible, the list of files in the archive is similar to the list of files in the installer.

From appveyor.bat excluded the command to edit the "version.h" file from the Vim source code.
When this pull request [#14465](https://github.com/vim/vim/pull/14465) is accepted from the main Vim branch, it will not be necessary, since the patch number will be inserted automatically.

Other minor changes.

### CODE REVIEW

File: appveyor.bat [hunk 1]
```diff
@@ -233,14 +233,17 @@ goto :eof
...
-type version.h
+@rem %CYGWIN_DIR%\bin\bash -lc "cd $(cygpath '%APPVEYOR_BUILD_FOLDER%')/vim/src && ../../scripts/patchlevel.sh"
+@rem call ..\..\scripts\patchlevel.cmd
+@rem type version.h
 
 :: Build GUI version
-nmake -f Make_mvc.mak ^
+nmake.exe -f Make_mvc.mak ^
 	GUI=yes OLE=yes DIRECTX=yes ^
-	FEATURES=HUGE IME=yes MBYTE=yes ICONV=yes DEBUG=no ^
```
Comment (chrisbra) [inline appveyor.bat:243]: why do you remove `MBYTE`?

File: appveyor.bat [hunk 2]
```diff
@@ -250,9 +253,9 @@ nmake -f Make_mvc.mak ^
 	TERMINAL=yes SODIUM=%SODIUM_DIR% ^
 	|| exit 1
 :: Build CUI version
-nmake -f Make_mvc.mak ^
+nmake.exe -f Make_mvc.mak ^
 	GUI=no OLE=no DIRECTX=no ^
-	FEATURES=HUGE IME=yes MBYTE=yes ICONV=yes DEBUG=no ^
```
Comment (chrisbra) [inline appveyor.bat:255]: same here

File: appveyor.bat [hunk 3]
```diff
@@ -318,23 +326,27 @@ copy /Y %DEPENDENCIES%\gettext64\libintl-8.dll       ..\runtime\GvimExt64\
...
+@rem if exist c:\gettext%BIT%\libgcc_s_sjlj-1.dll copy /Y c:\gettext%BIT%\libgcc_s_sjlj-1.dll ..\runtime\
 copy /Y winpty* ..\runtime\
 copy /Y winpty* ..\..\
-set dir=vim%TAG_NAME:~1,1%%TAG_NAME:~3,1%
+set "dir=vim%TAG_NAME:~1,1%%TAG_NAME:~3,1%"
 mkdir ..\vim\%dir%
-xcopy ..\runtime ..\vim\%dir% /Y /E /V /I /H /R /Q
+xcopy ..\runtime ..\vim\%dir% /Y /E /V /I /H /R /Q /EXCLUDE:..\..\exclist.txt
 :: need to escape the ! because of delayedexpansion being active
-7z a ..\..\gvim_%TAG_NAME:~1%_%ARCH%.zip ..\vim -x^^!vim\%dir%\syntax\testdir\
```
Comment (chrisbra) [inline appveyor.bat:334]: why did you remove the syntax\testdir? ah, its in the exclude list

File: appveyor.bat [hunk 4]
```diff
@@ -318,23 +326,27 @@ copy /Y %DEPENDENCIES%\gettext64\libintl-8.dll       ..\runtime\GvimExt64\
...
+7z.exe a ..\..\gvim_%TAG_NAME:~1%_%ARCH%.zip ..\vim
 
 :: Create installer
-%CYGWIN_DIR%\bin\bash -lc "cd $(cygpath '%APPVEYOR_BUILD_FOLDER%')/vim/runtime/doc && touch ../../src/auto/config.mk && make uganda.nsis.txt"
+@rem %CYGWIN_DIR%\bin\bash -lc "cd $(cygpath '%APPVEYOR_BUILD_FOLDER%')/vim/runtime/doc && touch ../../src/auto/config.mk && make uganda.nsis.txt"
```
Comment (chrisbra) [inline appveyor.bat:345]: > touch ../../src/auto/config.mk

I think this is still needed

File: appveyor.bat (hunk 1 above)
Comment (RestorerZ) [reply appveyor.bat:243]: This is no longer required, it was used [earlier](https://github.com/vim/vim-win32-installer/pull/336#issuecomment-2011765874). This definition is now always set to "yes".

> MBYTE is an old option which has been removed and always enabled now.

File: appveyor.bat (hunk 3 above)
Comment (RestorerZ) [reply appveyor.bat:334]: Yes.

File: appveyor.bat (hunk 4 above)
Comment (RestorerZ) [reply appveyor.bat:345]: File "config.mk" it is used only in Unix-like systems. It is not required on Windows.
Now "nmake.exe" is used everywhere.

Comment (RestorerZ) [reply appveyor.bat:345]: I've just looked through the appveroy logs — everything is more or less normal.
Only for the 32‐bit version, there is one failure in the tests.

> Failures: 
	From test_options.vim:
	Found errors in Test_shortmess_F3():
	command line..script C:\projects\vim-win32-installer\vim\src\testdir\runtest.vim[607]..function RunTheTest[57]..Test_shortmess_F3 line 19: Expected ['bar'] but got ['foo']
TEST FAILURE 
NMAKE : fatal error U1077: 'if exist test.log ( echo TEST FAILURE & exit /b 1 )  else ( echo ALL DONE )' : return code '0x1'
Stop

Comment (RestorerZ) [reply appveyor.bat:345]: And this is also for the 64‐bit version.

> highlight.c(882): warning C4267: '+=': conversion from 'size_t' to 'int', possible loss of data

File: appveyor.bat (hunk 1 above)
Comment (chrisbra) [reply appveyor.bat:243]: I see, thanks

File: appveyor.bat (hunk 4 above)
Comment (chrisbra) [reply appveyor.bat:345]: unrelated issue, should have been fixed by: https://github.com/vim/vim/commit/72a156b4edfdf1502d421370cf38281c258e3c4a

**Comment by chrisbra** [comment]
okay thanks. Let me merge it.
